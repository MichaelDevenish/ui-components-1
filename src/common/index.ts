export { Props } from './types'
export { getColor } from './legacy'
