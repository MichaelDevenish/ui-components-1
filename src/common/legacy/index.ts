export { getColor } from './colors/getColor'
