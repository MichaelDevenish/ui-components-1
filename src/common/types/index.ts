/**
 * For Examples, see:
 * https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts
 * https://github.com/alm-tools/alm/blob/master/src/common/types.ts
 */
import { Size as _s, AllSizes } from './size'

namespace Props {
  export import Size = _s
}

export {
  Props,
  AllSizes
}
