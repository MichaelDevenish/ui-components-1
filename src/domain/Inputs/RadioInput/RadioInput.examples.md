#### Basic Radio Input

 ```jsx
initialState = { value: 'No' };

<div>
  <RadioInput
   name='yes'
   label='Yes'
   value='Yes'
   isChecked={state.value === 'Yes'}
   onChange={(value) => setState({value})}
  />
  <RadioInput
   name='no'
   label='No'
   value='No'
   isChecked={state.value === 'No'}
   onChange={(value) => setState({value})}
  />
  <RadioInput
    isDisabled
    name='no-comment'
    label='No comment (disabled radio)' 
    isChecked={state.value === 'No comment'}
    onChange={(value) => setState({value})}
  />
</div>
```

#### Radio Input Button

```jsx
initialState = { button: 'No' };

<div>
  <RadioInput
   isButton
   name='button-yes'
   label='Yes'
   value='Yes'
   isChecked={state.button === 'Yes'}
   onChange={(value) => setState({button: value})}
  />
  <RadioInput
   isButton
   name='button-no'
   label='No'
   value='No'
   isChecked={state.button === 'No'}
   onChange={(value) => setState({button: value})}
  />
  <RadioInput
    isButton
    isDisabled
    name='button-no-comment'
    label='No comment (disabled radio)' 
    isChecked={state.button === 'No comment'}
    onChange={(value) => setState({button: value})}
  />
</div>
```

 #### Radio Input with JSX

 ```jsx
import { FontAwesomeIcon } from '@Domain/Icons';
import { Emoji } from '@Domain/Typographies';

initialState = { star: 'Yes' };

<div>
  <RadioInput
   name='star-yes'
   label={<div><FontAwesomeIcon type='star' /> I have a star!</div>}
   value='Yes'
   isChecked={state.star === 'Yes'}
   onChange={(value) => setState({star: value})}
  />
  <RadioInput
   name='star-no'
   label={<div><Emoji emoji='cry' /> I don't have a star </div>}
   value='No'
   isChecked={state.star === 'No'}
   onChange={(value) => setState({star: value})}
  />
</div>
```

 #### Radio Input with HandleBlur

 ```jsx
initialState = { value: 'No' };

<div>
  <RadioInput
   handleBlur={() => {setState({value:''}); alert('blur value')}} 
   name='special-yes'
   label='Yes'
   value='Yes'
   isChecked={state.value === 'Yes'}
   onChange={(value) => setState({value})}
  />
  <RadioInput
     handleBlur={() => {setState({value:''}); alert('blur value')}} 
     name='special-no'
     label='No'
     value='No'
     isChecked={state.value === 'No'}
     onChange={(value) => setState({value})}
    />
</div>
```
