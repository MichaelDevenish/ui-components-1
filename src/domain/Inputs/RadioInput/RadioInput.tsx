import React from 'react'
import { IGenericInputProps, Input } from '../Input'
import classNames from 'classnames'

const style = require('./style.scss')

export interface IRadioInputProps extends IGenericInputProps {
  /** Label to display next to the radio */
  label: JSX.Element | string
  /** If true, the radio input is wrapped with a button */
  isButton?: boolean
}

export class RadioInput extends React.PureComponent<IRadioInputProps> {
  public static defaultProps: Partial<IRadioInputProps> = {
    isButton: false,
    isDisabled: false
  }

  get infoLabel (): JSX.Element | null {
    const {
      id,
      name,
      label,
      isButton
    } = this.props

    if (!label) {
      return null
    }

    return (
      <label
        htmlFor={id || name}
        className={classNames('radio', { 'radio-button': isButton })}
      >
        {label}
      </label>
    )
  }

  public render (): JSX.Element {
    const {
      className,
      ...props
    } = this.props

    return (
      <div
        className={classNames( style.radioInput, className)}
      >
        <Input
          {...props}
          type='radio'
        />
        {this.infoLabel}
      </div>
    )
  }
}
