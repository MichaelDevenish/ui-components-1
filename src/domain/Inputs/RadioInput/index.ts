export { RadioInput, IRadioInputProps } from './RadioInput'
