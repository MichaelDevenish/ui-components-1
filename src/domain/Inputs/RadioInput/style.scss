@import '../../../common/sass/variables';

:local(.radioInput) {
  label {
    color: $n800;
    cursor: default;
    display: block;
    font-size: $fz-small;
    line-height: $lh-small;
    margin-bottom: 12px;

    &.horizontal-radio-group {
      display: inline-block;
      margin-right: 32px;
    }

    span {
      &::before {
        background-color: $n150;
        border: 2px solid $n400;
        border-radius: 50%;
        content: '';
        display: inline-block;
        height: 16px;
        margin-right: 12px;
        vertical-align: middle;
        width: 16px;
      }
    }

    input {
      display: none;

      &:hover {
        & + span {
          &::before {
            background-color: $n300;
            cursor: pointer;
          }
        }

        &:disabled + span {
          &::before {
            background-color: $n150;
            border-color: $n300;
            cursor: not-allowed;
          }
        }
      }

      &:checked {
        & + span {
          &::before {
            background-image: radial-gradient(circle closest-side, $i400 , $i400 50%, transparent 50%, transparent 100%);
            border-color: $i400;
            cursor: pointer;
          }
        }

        &:disabled + span {
          &::before {
            background-image: radial-gradient(circle closest-side, $n300 , $n300 50%, transparent 50%, transparent 100%);
            border-color: $n300;
            cursor: not-allowed;
          }
        }
      }

      &:disabled {
        & + span {
          color: $n500;

          &::before {
            background-color: $n150;
            border-color: $n300;
            cursor: not-allowed;
          }
        }

      }

    }
  }
}
