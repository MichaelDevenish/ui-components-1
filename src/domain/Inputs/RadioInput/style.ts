import styled from 'styled-components'

import { Props } from '../../../common'
import { styleForMargins } from '../../Spacers/services/margins'

export interface IStyledRadioInputWrapperProps {
  margins?: Props.IMargins
}

const StyledRadioInput = styled.input`
  line-height: 16px;

  margin: 0;

  &::-ms-clear {
    display: none;
  }
`

const StyledRadioInputWrapper = styled.div`
  ${(props: IStyledRadioInputWrapperProps) => styleForMargins(props.margins)}
`

export {
  StyledRadioInput,
  StyledRadioInputWrapper
}
