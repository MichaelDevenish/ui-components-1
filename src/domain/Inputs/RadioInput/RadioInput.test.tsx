import { shallow } from 'enzyme'
import React from 'react'

import { RadioInput } from './RadioInput'
import { Props } from '../../../common'

const dummyClick = () => console.log('hey')

const exampleOptions =[{
    label:'123123',
    value:'option 1',
    isDisabled:false
  },
  {
    label:'this is option 2 (I am disabled)',
    value:'option 2',
    isDisabled:true
  },
  {
    label:'this is option 3',
    value:'option 3',
    isDisabled:false
  },
  {
    label:'this is option 4',
    value:'option 4',
    isDisabled:false
  }]

describe('<RadioInput />', () => {
  it(`should render a radio input`, () => {
    const wrapper = shallow(
      <RadioInput
        id='key'
        handleChange={dummyClick}
        options={exampleOptions}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it(`should render a radio input with horizontal buttons`, () => {
    const wrapper = shallow(
      <RadioInput
        id='key'
        handleChange={dummyClick}
        options={exampleOptions}
        orientation={Props.Orientation.Horizontal}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
