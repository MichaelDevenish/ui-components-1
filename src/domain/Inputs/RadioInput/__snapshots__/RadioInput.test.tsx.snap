// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioInput /> should render a radio input with a component as the label 1`] = `
<styled.div>
  <div
    className="radioInput"
  >
    <styled.input
      className="input "
      data-component-type="radio_input"
      disabled={false}
      id="test-jsx-input"
      name="test-jsx-input"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="radio"
      htmlFor="test-jsx-input"
    >
      <FontAwesomeIcon
        type="star"
      />
    </label>
  </div>
</styled.div>
`;

exports[`<RadioInput /> should render a radio input with button 1`] = `
<styled.div>
  <div
    className="radioInput"
  >
    <styled.input
      className="input "
      data-component-type="radio_input"
      disabled={false}
      id="test-button-input"
      name="test-button-input"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="radio radio-button"
      htmlFor="test-button-input"
    >
      Hey I am a test string
    </label>
  </div>
</styled.div>
`;

exports[`<RadioInput /> should render a radio input without button 1`] = `
<styled.div>
  <div
    className="radioInput"
  >
    <styled.input
      className="input "
      data-component-type="radio_input"
      disabled={false}
      id="test-input"
      name="test-input"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="radio"
      htmlFor="test-input"
    >
      Hey I am a test string
    </label>
  </div>
</styled.div>
`;
